### PR TITLE
readme: add note about uncaught exceptions

### PR DESCRIPTION
**Pull request by arve0**
fixes issue #917

### CODE REVIEW

**Comment by jamestalmage** [comment]
I'm not sure this is the correct behavior. We may just want to run `after.always` on uncaught exceptions

**Comment by novemberborn** [comment]
> I'm not sure this is the correct behavior. We may just want to run after.always on uncaught exceptions

Uncaught exceptions cause the worker to be torn down. That seems reasonable to me. The behavior is somewhat similar to `--bail`, which we've previously decided does not cause `after.always` to run.

File: readme.md [hunk 1]
```diff
@@ -433,7 +433,7 @@ test.failing('demonstrate some bug', t => {
 
 AVA lets you register hooks that are run before and after your tests. This allows you to run setup and/or teardown code.
 
-`test.before()` registers a hook to be run before the first test in your test file. Similarly `test.after()` registers a hook to be run after the last test. Use `test.after.always()` to register a hook that will **always** run once your tests and other hooks complete. `.always()` hooks run regardless of whether there were earlier failures, so they are ideal for cleanup tasks.
+`test.before()` registers a hook to be run before the first test in your test file. Similarly `test.after()` registers a hook to be run after the last test. Use `test.after.always()` to register a hook that will **always** run once your tests and other hooks complete. `.always()` hooks run regardless of whether there were earlier failures, so they are ideal for cleanup tasks. **Note**: In the case of uncaught exceptions, `test.after.always()` will not be called.
```
Comment (novemberborn) [inline readme.md]: I would rephrase this though:

> … they are ideal for cleanup tasks. There are two exceptions to this however. If you use `--fail-fast` AVA will stop testing as soon as a failure occurs, and it won't run any hooks including the `.always()` hooks. Uncaught exceptions will crash your tests, possibly preventing `.always()` hooks from running.

Comment (arve0) [inline readme.md]: Uncaught exceptions will crash your tests, ~~possibly~~ preventing .always() hooks from running?

Comment (arve0) [inline readme.md]: I'm not familiar with the internals, but are there cases where uncaught exceptions will not prevent `.after` to run?

Comment (novemberborn) [inline readme.md]: Yes, if they occur after `.after` has run. With asynchronous tests it's a race between a test completing and an uncaught exception occurring.

**Comment by arve0** [comment]
Updated to new wording.

**Comment by novemberborn** [comment]
LGTM 😉

**Comment by sindresorhus** [comment]
LGTM, because it reflects reality.

Though, I'm not totally happy about `.always` not actually being always, but I think the earlier arguments were that you'd want the tests and fixtures to be in a consistent state so you can debug. Personally, I find it a bit annoying having to manually cleanup the leftovers. There's also the argument that cleanup should happen on startup, but then you get a dirty git repo. Feels like there's something we could do to ease this for the user.

**Comment by jamestalmage** [comment]
> There's also the argument that cleanup should happen on startup, but then you get a dirty git repo.

I usually get around that with `.gitignore`.

>  Feels like there's something we could do to ease this for the user.

What if we allowed a `.cleanup` modifier that ran before and after the tests?

Or what if we allowed an `and` modifier:

```js
test.before.and.after(fn)
test.beforeEach.and.after.always(fn)
```

**Comment by sindresorhus** [comment]
@jamestalmage Can you move the bottom part to a new issue?
